### PR TITLE
Full Pipeline

### PR DESCRIPTION
**Pull request by bh2smith**
1. Implement the following method 
```ts
async function internalizedTokenImbalance(
  txData: MinimalTxData,
  db: Queryable,
  simulator: TransactionSimulator
): Promise<void> 
```

Which takes a transaction, database and transaction simulator to perform the full pipeline process outlined in this [spec doc](https://docs.google.com/document/d/15mByvoI7EQ-wFAM9QkDCMnjA4AVjbG0_G0GdirziGBM/edit#) and illustrated in the screenshot below

2. Write some tests for it.
3. Update the index.ts file (i.e. the tenderly web-hook) to make use of this function.

Screenshot 2023-04-28 at 13 10 12

Concludes Phase 1 of part #177 (excluding a few follow ups).

### CODE REVIEW

**Comment by gentrexha** [comment]
Is this ready to be reviewed?

**Comment by bh2smith** [comment]
> Is this ready to be reviewed?

Almost -- I need to add that TODO test (but I already added it in a follow up PR). Have not had time to come back to this.

**Comment by gentrexha** [comment]
> > Is this ready to be reviewed?
> 
> Almost -- I need to add that TODO test (but I already added it in a follow up PR). Have not had time to come back to this.

Cool. No hurry. Just remove the Draft once ready and I'll take a look at it.

File: internal_transfers/actions/src/pipeline.ts
Comment (gentrexha) [inline internal_transfers/actions/src/pipeline.ts:21]: Out of curiosity: Do these logs get saved anywhere?

File: internal_transfers/actions/src/pipeline.ts
Comment (gentrexha) [inline internal_transfers/actions/src/pipeline.ts]: Nit: move comment before the if
```suggestion
    // If there is a simulation, get imbalances otherwise assume none.
    if (settlementSimulations) {
```

File: internal_transfers/actions/tests/e2e/pipeline.spec.ts
Comment (gentrexha) [inline internal_transfers/actions/tests/e2e/pipeline.spec.ts]: Aren't we always expecting the same thing here? Shouldn't the expected results of these tests differ?

File: internal_transfers/actions/tests/e2e/pipeline.spec.ts
Comment (gentrexha) [inline internal_transfers/actions/tests/e2e/pipeline.spec.ts:14]: I thought we removed this in an earlier [PR](https://github.com/cowprotocol/solver-rewards/pull/268), no?

**Review by gentrexha — APPROVED** [review]
LGTM. I still recommend waiting for a review from someone on the front-end team for more TS-specific comments.

I left some questions.

**Review by gentrexha — APPROVED** [review]
LGTM. I still recommend waiting for a review from someone on the front-end team for more TS-specific comments.

I left some questions.

File: internal_transfers/actions/src/pipeline.ts
Comment (bh2smith) [reply internal_transfers/actions/src/pipeline.ts]: What if I just remove the first half of the sentence since we already said if and then keep the comment where it is so it reads like a sentence? 

If ... get

File: internal_transfers/actions/tests/e2e/pipeline.spec.ts
Comment (bh2smith) [reply internal_transfers/actions/tests/e2e/pipeline.spec.ts]: I think I forgot to put "expect" statements.

File: internal_transfers/actions/tests/e2e/pipeline.spec.ts
Comment (bh2smith) [reply internal_transfers/actions/tests/e2e/pipeline.spec.ts:14]: Oh. This DB is totally different from that one. This mini project is entirely self contained and independent of solver payouts.

File: internal_transfers/actions/src/pipeline.ts
Comment (bh2smith) [reply internal_transfers/actions/src/pipeline.ts:21]: These logs appear in the web-action runs: [Here is an example](https://dashboard.tenderly.co/gp-v2/solver-slippage/action/4f657ddd-c054-4704-9e1e-50385ad2fc2c/execution/772d2a71-47e9-4844-a19c-91ff603e1d46) if you have tenderly credentials.

File: internal_transfers/actions/tests/e2e/pipeline.spec.ts
Comment (bh2smith) [reply internal_transfers/actions/tests/e2e/pipeline.spec.ts]: Fixed this by adding assert Not Throw in two places and Assert throws in the third test.

Reason I "forgot" these was that the function doesn't return anything so there was nothing to expect being returned

File: internal_transfers/actions/src/accounting.ts
Comment (alfetopito) [inline internal_transfers/actions/src/accounting.ts]: Mutating input params? 😰

File: internal_transfers/actions/tests/e2e/pipeline.spec.ts
Comment (alfetopito) [inline internal_transfers/actions/tests/e2e/pipeline.spec.ts:14]: Nitpick: use a const/env var for the db URL

**Review by alfetopito — APPROVED** [review]
Looks mostly ok, just 2 comments

File: internal_transfers/actions/src/accounting.ts
Comment (bh2smith) [reply internal_transfers/actions/src/accounting.ts]: I can rework this.... 😄

File: internal_transfers/actions/tests/e2e/pipeline.spec.ts
Comment (bh2smith) [reply internal_transfers/actions/tests/e2e/pipeline.spec.ts:14]: but this is just a test... I would then be creating one additional unnecessary constant here.

Comment (bh2smith) [reply internal_transfers/actions/tests/e2e/pipeline.spec.ts:14]: Fixed [003b4a1](https://github.com/cowprotocol/solver-rewards/pull/261/commits/003b4a18ef0dca10c2b20a717998a0ac86c7216c)

File: internal_transfers/actions/src/accounting.ts
Comment (bh2smith) [reply internal_transfers/actions/src/accounting.ts]: Fixed [003b4a1](https://github.com/cowprotocol/solver-rewards/pull/261/commits/003b4a18ef0dca10c2b20a717998a0ac86c7216c)

**Pull request merged**
